src/main/java/fr/sncf/osrd/railjson/parser/RailScriptExprParser.java

```diff
@@ -240,6 +258,24 @@ private static int parseEnumVal(Enum<?>[] enumValues, String val) throws Invalid
         throw new InvalidInfraException("unknown enum value: " + val);
     }
 
+    private RSExpr.OptionalMatch<?> parseOptionalMatch(RJSRSExpr expr) throws InvalidInfraException {
+        var optionalMatchExpr = (RJSRSExpr.OptionalMatch) expr;
+        var name = optionalMatchExpr.name;
+        if (varTypes.containsKey(name))
+            throw new InvalidInfraException("Duplicate optional match name " + name);
+        var noneExpr = parse(optionalMatchExpr.caseNone);
+        var contentExpr = parseOptionalExpr(optionalMatchExpr.expr);
+        var type = contentExpr.getType(argTypes);
+        var subType = type.subType();
+        varTypes.put(name, subType);
+        var someExpr = parse(optionalMatchExpr.caseSome);
+        varTypes.remove(name);
+
+        @SuppressWarnings({"unchecked", "rawtypes"})
+        var res = new RSExpr.OptionalMatch(contentExpr, noneExpr, someExpr, name);
+        return res;
```

Comment:
```suggestion
        return new RSExpr.OptionalMatch(contentExpr, noneExpr, someExpr, name);
```

Reply:
Actually this is required to make the `SuppressWarning` work from the line above.